### ISSUE TITLE
Polish reparseable entities

### ISSUE THREAD

**Issue by hurricup** [Camelcade/Perl5-IDEA#2259]
- [x] make regex token reparseable
- [x] make attributes reparseable
- [x] make string content in attributes reparseable
- [x] test and forbid reparsing if template part is in the block (except the POD)
- [x] check balance of not on only braces, but parens and brackets. Orphan close is ok, orphan open is not.
- [x] make anon hashes reparseable
- [x] make anon arrays reparseable
- [x] make parenthesized expressions reparseable
- [x] make sub signatures reparseable (may change to prototype, lexer confirmation?)
- [x] make method signatures reparseable (lexer confirmation?)
- [x] make all strings reparseable
- [x] make regular expression expressions reparseable (from start to end)
- [x] make qw list reparseable, 
- [x] make qw and regexes  sublexed all the time
- [x] make dereferences, including interpolations reparsable (may convert to a braced variable name)
- [x] add highlighting lexer recovery test (show number of recovery points, offsets and longerst range)
- [x] line comments can be reparseable leafs 
- [x] variable names can be reparseable leafs
- [x] spaces could be reparseable, but not sure, also hard to handle (require platform implementation)
- [x] block comment may be reparseable?
- [x] subs names are too hard. they may become package or resword or something
- [x] subs names in dereference may be reparseable
- [x] packages may be reparseable in certain places, like namespace name, `use`, `my`/`our`/etc
- [x] qualifying packages may be reparseable if they are end with `::`
- [x] string content in bare string may be reparseable
- [x] simple q strings contents  may be reparseable
- [x] simple qq strings contents  may be reparseable
- [x] simple qx strings contents may be reaprseable
- [x] hash index can be reparseable
- [x] array index can be reparseable
- [x] parenthesized call arguments (need to get rid of meta rules)
- [x] make template element reparseable for embedded perl
- [x] make template element reparseable for mojo templates
- [x] make template element reparseable for mason2 templates
- [x] make template element reparseable for html::mason templates
- [x] make some parts of POD reparseable (separate issue? )
- [x] make some parts of Template Toolkit reparseable (separate issue?)
- [x] methods and modifiers should probably be reparseable too?
- [x] We probably can reparse any statement. Take new text. Wrap with dumb code lines. Parse, check if a statement is in the middle of the 3 kids.
- [x] We probably should check if heredoc body is at first line of the block, e.g.:
```
<<EOM, {
   blabla
EOM
   say 'hi'
}
```

Working on leaf reparsing makes sense only after platform adjustments. Atm. we can't reparse typing token when lexed in the end.
